nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -318,26 +324,42 @@ bool CollisionMonitor::configureSources(
         rclcpp::ParameterValue("scan"));  // Laser scanner by default
       const std::string source_type = get_parameter(source_name + ".type").as_string();
 
+      nav2_util::declare_parameter_if_not_declared(
+        node, source_name + ".source_timeout",
+        rclcpp::ParameterValue(source_timeout.seconds()));    // node source_timeout by default
+      const rclcpp::Duration sensor_specific_source_timeout = rclcpp::Duration::from_seconds(
+        get_parameter(
+          source_name + ".source_timeout").as_double());
+
+      nav2_util::declare_parameter_if_not_declared(
+        node, source_name + ".block_if_invalid",
+        rclcpp::ParameterValue(block_if_invalid));    // node block_if_invalid by default
+      const bool sensor_specific_block_if_invalid =
+        get_parameter(source_name + ".block_if_invalid").as_bool();
+
```

Comment:
Please move each sources' timeout and `ignore_invalid` (or whatever you call it) logic inside `Source::getCommonParameters()` since both parameters belong to their own data source

Reply:
I think he's also made this so that there can be a global version as well, not just per-sensor if they want to specify in the parent namespace for all

Reply:
Indeed

Reply:
I still inclined to move this logic inside `Source::getCommonParameters()`: by its design all parameters belonging to any specific source or polygons should be declared/get in its routines. We could construct the source with common for all sources timeout, and then later on `configure()` stage (when getCommonParameters is being called), re-assign the `source_timeout_` class' value for specific for its source value. I think, this will be more clear to not making the part of code outside its handling class.